params/config.go

```diff
@@ -484,6 +486,19 @@ var (
 	TestRules = TestChainConfig.AvalancheRules(new(big.Int), 0)
 )
 
+// UpgradeConfig includes the following configs that may be specified in upgradeBytes:
+// - Timestamps that enable avalanche network upgrades,
+// - Enabling or disabling precompiles as network upgrades.
+type UpgradeConfig struct {
+	// Config for enabling and disabling precompiles as network upgrades.
+	PrecompileUpgrades []PrecompileUpgrade `json:"precompileUpgrades,omitempty"`
```

Comment:
Yes it's activated with the DUpgrade in `vm.go`. I thought it was easier to keep Coreth and Subnet-EVM closer in line rather than having a different mechanism.

Additionally, we can't import the Warp precompile in the `params/` package, so it needs to be done externally anyways. Open to suggestion, but after playing around a bit, I thought the best path was to simply share the same code with Subnet-EVM and set the `PrecompileUpgrades` field in `vm.go` here: https://github.com/ava-labs/coreth/blob/migrate-warp-precompile/plugin/evm/vm.go#L470

Reply:
I was just in favor of having less code to reduce risk. But I think that's fine.